### ISSUE TITLE
Batch uploaded GPX-files in a zip-file, not showing up in Potlatch 2

### ISSUE THREAD

**Issue by magu01** [systemed/potlatch2#128]
I zipped a bunch of gpx-files as mentioned in this link.
https://wiki.openstreetmap.org/wiki/Batch_Upload_of_GPX_(various_tools)

But when i try to load the trace in Potlatch 2 I can't see the trace. Potlatch is not showing any trace.

This is the trace.
https://www.openstreetmap.org/user/magu01/traces/3161726

The API is allready in place and the problem was discused in this b

**Comment by magu01** [comment]
tomhughes
"I suspect the problem is that editors don't cope with the slightly odd way we return the data, which is basically by concatenating all the files, and they are at best only parsing the first XML document they see."

**Comment by systemed** [comment]
Yes, that's probably the case. The document that P2 fetches is http://www.openstreetmap.org/api/0.6/gpx/3161726/data.xml . This contains several XML documents appended together, and if you run it through an XML validator it'll squawk:

````
richard$ xmllint /Users/richard/Downloads/3161726.xml 
/Users/richard/Downloads/3161726.xml:378: parser error : XML declaration allowed only at the start of the document
<?xml version="1.0" encoding="UTF-8"?>
     ^
/Users/richard/Downloads/3161726.xml:379: parser error : Extra content at the end of the document
<gpx
^
````

Most probably the Flash XML parser is deciding that it's an invalid document and refusing to parse any of it.

**Comment by magu01** [comment]
Can you make a check if its muliple xmls and splitting it before sending it to the parser? If not, how do you suggest its solved?

**Comment by systemed** [comment]
Sure - the string could be split on loading and then each XML document processed individually. It's not something I'm likely to work on any time soon but I'll leave this open in case anyone else wants to. `process()` in `halcyon/connection/Trace.as` would be the place to start looking.

**Comment by magu01** [comment]
Thanks for the link! Sounds good that you keep the case open. A good feature if it works!
